### PR TITLE
[JME NanoAOD] Changes for event size reduction

### PR DESCRIPTION
**Pull request by nurfikri89**
#### PR description:

This PR is to reduce the size of custom JME NanoAODs. The following changes are made:
- Check that `npuProd[ibx].getPU_pT_hats()` size is more than 0 in `PhysicsTools/NanoAOD/plugins/NPUTablesProducer.cc`.
- Switch on storing Parton Shower (PS) weights.
- b-tagging algorithm outputs are saved for AK4 jets with pT > 15 GeV.
- Low-level input variables for PileUp jet ID and Quark-Gluon Likelihood are saved for AK4 jets with pT > 10 GeV.
- Remove AK4 PF and AK8 PF jet collections.
- Add separate AK8 GEN and AK8 Puppi jet collections for JEC studies. Minimal information are saved.
- For reclustered GEN jet collections, the minimum pT at the jet clustering level uses the default value (3 GeV) in `RecoJets/JetProducers/python/ak4GenJets_cfi.py`.
- For reclustered reco-level jet collections, the minimum pT at the jet clustering level is set to be 0. Then, only jets with (JEC-corrected) pT > 8 GeV are saved. Additionally for MC samples, jets with pT < 8 GeV are saved if they are matched to GEN jets.

With this PR, the event size is now reduced to **6.42 kb/event** from **9.32 kb/event**. The comparison was made using 10K events from a TTJets RunIISummer19UL17MiniAOD sample. The changes in this PR was discussed in the 13/01/2021 XPOG meeting [1]. 

[1] https://indico.cern.ch/event/978436/

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Once merged, this PR needs to be backported to 10_6_X,  for the ultra legacy campaigns.

### CODE REVIEW

**Comment by nurfikri89** [comment]
FYI @camclean @alefisico

**Comment by gouskos** [comment]
please test

**Comment by mariadalfonso** [comment]
please test workflow 25202.15,10224.15,11024.15

(to test JMEnano workflow)

**Comment by mariadalfonso** [comment]
@andrzejnovak
please confirm that this update is good too BTV as well

File: PhysicsTools/NanoAOD/python/custom_jme_cff.py [hunk 1]
```diff
@@ -18,118 +17,126 @@
...
+# - ak8gen (GenJetAK8 in NanoAOD), slimmedGenJetsAK8 in MiniAOD
+# Below is a list of genjets that we can save in NanoAOD. Set
 # "enabled" to true if you want to store the jet collection
 config_genjets = [
-  { 
-    "jet"     : "ak8gen",    
-    "enabled" : False, 
-  },  
+  {
+    "jet"     : "ak6gen",
```
Comment (gouskos) [inline PhysicsTools/NanoAOD/python/custom_jme_cff.py:55]: What's the motivation for the change from ak8 -> ak6 in the naming ?

Comment (nurfikri89) [reply PhysicsTools/NanoAOD/python/custom_jme_cff.py:55]: I had set the new AK8 GEN jet collection reclustering in a function [here](https://github.com/cms-sw/cmssw/blob/6e83ab4ee2d62986883fb4871cf4961ba420bd83/PhysicsTools/NanoAOD/python/custom_jme_cff.py#L952). The change in the line above is just to put a dummy example using ak6gen. The ak6gen jet clustering is switched off.

**Comment by andrzejnovak** [comment]
@mariadalfonso afaik btv is not using this JME Nano

**Comment by gouskos** [comment]
+xpog

All tests passed, and changes to the code are consistent with the PR description
Please prepare the corresponding backports for CMSSW_11_2_X and CMSSW_10_6_X

**Comment by nurfikri89** [comment]
@gouskos The backport PRs are #32759 (10_6_X) and #32760 (11_2_X)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/32722	6e83ab4 
- code changes are in line with the PR description; reco review covers just PhysicsTools/PatAlgos/python/tools/jetCollectionTools.py , which has some extension in functionality without changing the defaults
- jenkins tests pass and comparisons with the baseline show no relevant differences

**Comment by qliphy** [comment]
+1
